pytestqt/wait_signal.py

```
@@ -154,107 +282,226 @@ class MultiSignalBlocker(_AbstractSignalBlocker):
         marks that we finished because signals.
         """
-        if self.order == "none":
+        self._record_emitted_signal_if_possible(unique_signal, *args)
+
+        self._check_signal_match(unique_signal, *args)
+
+        if self._all_signals_emitted():
+            self.signal_triggered = True
+            try:
+                self._cleanup()
+            finally:
+                self._loop.quit()
+
+    def _record_emitted_signal_if_possible(self, unique_signal, *args):
+        if self._are_signal_names_available():
+            self.all_signals_and_args.append(
+                SignalAndArgs(signal_name=self._signal_names[unique_signal], args=args))
+
+    def _check_signal_match(self, unique_signal, *args):
+        if self._order == "none":
             # perform the test for every matching index (stop after the first one that matches)
-            successfully_emitted = False
-            successful_index = -1
-            potential_indices = self._get_unemitted_signal_indices(signal)
-            for potential_index in potential_indices:
-                if self._check_callback(potential_index, *args):
-                    successful_index = potential_index
-                    successfully_emitted = True
-                    break
-
-            if successfully_emitted:
+            try:
+                successful_index = self._get_first_matching_index(unique_signal, *args)
                 self._signals_emitted[successful_index] = True
-        elif self.order == "simple":
-            potential_indices = self._get_unemitted_signal_indices(signal)
-            if potential_indices:
-                if self._signal_expected_index == potential_indices[0]:
-                    if self._check_callback(self._signal_expected_index, *args):
-                        self._signals_emitted[self._signal_expected_index] = True
-                        self._signal_expected_index += 1
+            except:  # none found
```

Better to be explicit about which exceptions you plan to catch here... a bare except like this is considered somewhat dangerous because it will capture absolutely any exception